### DISCUSSION TITLE
Copy, then Close -- wipes the Paste buffer

### DISCUSSION THREAD

**Discussion by TechnoRobert** [audacity/audacity#4698] [General]
In recent releases of Audacity, I've found that if I open an audio file and copy something from it--but then close that file--then I cannot paste what I've copied. Instead, I must keep the file I've copied from open until I've completed the paste to the new location.

Perhaps there is a good reason for this new behavior. Can someone explain the good reason...or tell me how to propose that the copy/paste process revert to its previous behavior? 

(As a newbie here, I'm not positive that I did a thorough search for this topic, but upon typing the title, I got "There are no matching discussions.")

**Comment by Paul-Licameli** [comment]
This has been so since version 3.0.0 which reorganized project file storage to use the Sqlite database system instead of a large set of small files in a companion directory of the .aup file.

Lifting this limitation would require a temporary dB to hold copies of just the required data, while still letting the project database to close promptly and free up other resources.

This extra implementation difficulty was not judged to be worth the effort.

**Comment by TechnoRobert** [comment]
I appreciate your quick and comprehensive explanation, Paul. 

Guess I'll have to learn to live with it when working with audio, but all applications I've ever used for manipulating other data types--text, video, and image--normally allow copying from a file, closing the file, and then pasting to the new location. I mean, the whole concept of a "paste buffer" has historically been that whatever you copied went in there and would stay there until it got replaced by something else. The idea that some operation would clear the Paste buffer in the background is new...at least to me.

Even though Audacity has been behaving this way for the last two years, I still find myself occasionally slipping into the copy/close/paste habit that works fine in other media processing apps, and getting burned. ("Oops--gotta go back and retrieve the file I copied from, find the place I copied from, re-copy, don't close yet, go to new place and paste, now go back and close, now return to new place.")

Sorry to hear that the extra implementation difficulty was not judged to be worth the effort. However, I suppose now I can stop hoping that each new release might "fix" what I thought must have been an oversight.

Again, thanks for the fast response.

**Reply by LWinterberg** [reply]
Where would you intend to paste this data to if it was possible? Straight into the file browser? Into another app? Audacity again? Something else?

**Reply by TechnoRobert** [reply]
You ask, "Where would you intend to paste this data to if it was possible?". It is, of course, currently possible! But _only_ if you have not yet closed the file from which it was copied. That is, if you keep the source file open, you can paste multiple times...but the moment you close the source file, the paste buffer vanishes and your next Cmd-V (or Ctrl-V) does nothing.

And, "where?" For me, it would always be into another--or the same--file within Audacity. For example: A sound effect, extracted from a file where I had a lot of noises. Or, a musical sequence that I want to cause to recur in several places in a soundtrack I'm building.

I can't think of a case in which, for something copied from Audacity, it would make any sense to paste "into the file browser" or "Into another app"; not sure what you're imagining there.

**Reply by LWinterberg** [reply]
I wasn't sure either what to imagine there, it's just that a workflow of open audacity > copy > quit audacity > open audacity seems very strange to me.

**Reply by TechnoRobert** [reply]
Oh, sorry for the confusion. At no point do I quit Audacity. 

I'm speaking of opening and closing files within Audacity. It's not unusual for me to have six to ten files open at once in Audacity. Therefore, when I open one solely for the purpose of extracting (copying) a snippet, I'd like to then close it immediately, then open the destination file to paste the clip wherever it's going. The issue I'm raising is that the act of closing the source file will now (since the release of 3.0) clear the paste buffer.
